src/main/java/org/myrobotlab/audio/AudioProcessor.java

```diff
@@ -195,21 +203,52 @@ public AudioData play(AudioData data) {
             }
           }
 
-          // BooleanControl
-          // muteControl=(BooleanControl)source.getControl(BooleanControl.Type.MUTE);
-          /*
-           * if (volume == 0) { muteControl.setValue(true); }
-           */
-
-          // the buffer of raw data could be published from here
-          // if a reference of the service is passed in
-
           if (audioFile.isMute()) {
             // NoOp for a mute audioFile.
           } else {
+
+            if (cnt % sampleInterval == 0) {
+              float[] samples = new float[buffer.length / 2];
+
+              float lastPeak = 0f;
+
+              int b = buffer.length;
+              // convert bytes to samples here
+              for (int i = 0, s = 0; i < b;) {
+                int sample = 0;
+
+                sample |= buffer[i++] & 0xFF; // (reverse these two lines
+                sample |= buffer[i++] << 8; // if the format is big endian)
+
+                // normalize to range of +/-1.0f
+                samples[s++] = sample / 32768f;
+              }
+
+              float rms = 0f;
+              float peak = 0f;
+              for (float sample : samples) {
+
+                float abs = Math.abs(sample);
+                if (abs > peak) {
+                  peak = abs;
+                }
+
+                rms += sample * sample;
+              }
+
+              rms = (float) Math.sqrt(rms / samples.length);
+
+              if (lastPeak > peak) {
+                peak = lastPeak; 
+              }
+
+              lastPeak = peak;
+              audioFile.invoke("publishPeak", peak * (float) audioFile.getPeakMultiplier());

```

Comment:
what's the point of computing the rms if all you are doing is publishing the peak value?